### PR TITLE
Map packet codecs (24w03a)

### PR DESCRIPTION
**Pull request by MattiDragon**
Started the work. Still gonna map a bit more (notable recipe code).

### CODE REVIEW

File: mappings/net/minecraft/advancement/Advancement.mapping [hunk 1]
```diff
@@ -1,6 +1,7 @@
 CLASS net/minecraft/class_161 net/minecraft/advancement/Advancement
 	FIELD field_47179 CODEC Lcom/mojang/serialization/Codec;
 	FIELD field_47180 CRITERIA_CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_47996 PACKETT_CODEC Lnet/minecraft/class_9139;
```
Comment (Shnupbups) [inline mappings/net/minecraft/advancement/Advancement.mapping]:
```suggestion
	FIELD field_47996 PACKET_CODEC Lnet/minecraft/class_9139;
```

File: mappings/net/minecraft/network/PacketByteBuf.mapping [hunk 2]
```diff
@@ -342,6 +342,7 @@ CLASS net/minecraft/class_2540 net/minecraft/network/PacketByteBuf
 		COMMENT written
 		COMMENT @see #readNbt()
 		COMMENT @see #readNbt(NbtSizeTracker)
+		ARG 1 ntb
```
Comment (Shnupbups) [inline mappings/net/minecraft/network/PacketByteBuf.mapping]:
```suggestion
		ARG 1 nbt
```

File: mappings/net/minecraft/network/PacketByteBuf.mapping [hunk 3]
```diff
@@ -1072,6 +1073,29 @@ CLASS net/minecraft/class_2540 net/minecraft/network/PacketByteBuf
...
+		ARG 1 vector
+	METHOD method_56340 readNbt (Lio/netty/buffer/ByteBuf;Lnet/minecraft/class_2505;)Lnet/minecraft/class_2520;
+		ARG 0 buf
+		ARG 1 sizeTracker
+	METHOD method_56341 writeNbt (Lio/netty/buffer/ByteBuf;Lnet/minecraft/class_2520;)V
+		ARG 0 buf
+		ARG 1 nbt
+	METHOD method_56342 readVector3f (Lio/netty/buffer/ByteBuf;)Lorg/joml/Vector3f;
+		ARG 0 buf
+	METHOD method_56343 readQuanternionf (Lio/netty/buffer/ByteBuf;)Lorg/joml/Quaternionf;
```
Comment (Shnupbups) [inline mappings/net/minecraft/network/PacketByteBuf.mapping]:
```suggestion
	METHOD method_56343 readQuaternionf (Lio/netty/buffer/ByteBuf;)Lorg/joml/Quaternionf;
```

File: mappings/net/minecraft/network/codec/PacketCodecs.mapping [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	FIELD field_48550 VAR_INT Lnet/minecraft/class_9139;
+	FIELD field_48551 VAR_LONG Lnet/minecraft/class_9139;
+	FIELD field_48552 FLOAT Lnet/minecraft/class_9139;
+	FIELD field_48553 DOUBLE Lnet/minecraft/class_9139;
+	FIELD field_48554 STRING Lnet/minecraft/class_9139;
+	FIELD field_48555 NBT_ELEMENT Lnet/minecraft/class_9139;
+	FIELD field_48556 NBT_COMPUND Lnet/minecraft/class_9139;
+	FIELD field_48557 OPTIONAL_NBT Lnet/minecraft/class_9139;
+	FIELD field_48558 VECTOR3F Lnet/minecraft/class_9139;
+	FIELD field_48559 QUANTERNION Lnet/minecraft/class_9139;
```
Comment (Shnupbups) [inline mappings/net/minecraft/network/codec/PacketCodecs.mapping]:
```suggestion
	FIELD field_48559 QUATERNION Lnet/minecraft/class_9139;
```

**Review by Shnupbups — CHANGES_REQUESTED** [review]
some typos

File: mappings/net/minecraft/network/packet/s2c/config/ReadyS2CPacket.mapping [hunk 5]
```diff
@@ -1 +1,3 @@
 CLASS net/minecraft/class_8733 net/minecraft/network/packet/s2c/config/ReadyS2CPacket
+	FIELD field_48690 INSTANCE Lnet/minecraft/class_8733;
+	FIELD field_48691 ODEC Lnet/minecraft/class_9139;
```
Comment (Shnupbups) [inline mappings/net/minecraft/network/packet/s2c/config/ReadyS2CPacket.mapping]:
```suggestion
	FIELD field_48691 CODEC Lnet/minecraft/class_9139;
```

Comment (MattiDragon) [reply mappings/net/minecraft/network/packet/s2c/config/ReadyS2CPacket.mapping]: This one's caused by me using a macro on the packet codecs and it sometimes misfireing

**Review by Shnupbups — APPROVED** [review]

**Pull request merged**